### DISCUSSION TITLE
What config setting are required for change filament?

### DISCUSSION THREAD

**Discussion by Woodcat64** [Jyers/Marlin#30] [Q&A] [answered]
What kind of configuration requirements are needed to enable the change filament option in firmware?

**Comment by Jyers** [comment]
I'm assuming you mean in relation to the precompiled firmware. The reason it is disabled is because the filament change option requires that the distance to load/unload the filament to be configured in the configuration files. This distance can vary highly depending on Bowden tube length or extruder setup, hence why it can't be enabled in the precompiled options.

**Reply by DeNohvo** [reply]
Does this work without a sensor if i was to complie my own FW with correct distance etc?

**Reply by t3chguy** [reply]
Yes, those settings are used even if you have a runout sensor, that's how Marlin's M600 works.

**Reply by DeNohvo** [reply]
I may have misunderstood the feature, was thinking it is filimant run out feature!

If i understand correctly, it is if i want to change colours during a print?

**Comment by Jyers** [comment, marked as answer]
If you compile your own, all the settings you need can be found in the advanced pause section of configuration_adv.h

**Reply by jjfs127** [reply]
Can you elaborate on how this works with respect to temperature, must you manually set a temp or is there a setting per filament type?

**Reply by Jyers** [reply]
I have not fully looked into what defines the temperature it preheats to. I will have to look into the code for M600 and see if I can find the source for it.

**Reply by t3chguy** [reply]
It is controlled by https://github.com/MarlinFirmware/Configurations/blob/import-2.0.x/config/default/Configuration.h#L565

**Reply by Jyers** [reply]
That's what I thought but I wasn't sure. Thanks for the confirmation!
